EIPS/eip-721.md

```diff
@@ -0,0 +1,404 @@
+	function transferFrom(address _from, address _to, uint256 _deedId) external payable;
+
+    /// @notice Set or reaffirm the approved address for a deed
+    /// @dev The zero address indicates there is no approved address.
+    /// @dev Throws unless `msg.sender` is the current deed owner, or an authorized
+    ///  operator of the current deed owner.
+    /// @param _approved The new approved deed controller
+    /// @param _deedId The deed to approve
+    function approve(address _approved, uint256 _deedId) external payable;
+
+    /// @notice Enable or disable approval for a third party ("operator") to manage
+    ///  all your deeds.
+    /// @dev Emits the ApprovalForAll event
+    /// @param _operator Address to add to the set of authorized operators.
+    /// @param _approved True if the operators is approved, false to revoke approval
+    function setApprovalForAll(address _operator, bool _approved) external;
+
+    /// @notice Get the approved address for a single deed
+    /// @dev Throws if `_deedId` is not a valid deed
+    /// @param _deedId The deed to find the approved address for
+    /// @return The approved address for this deed, or the zero address if there is none
+    function getApproved(uint256 _deedId) returns (address);
+
+    /// @notice Query if an address is an authorized operator for another address
+    /// @param _owner The address that owns the deeds
+    /// @param _operator The address that acts on behalf of the owner
+    /// @return True if `_operator` is an approved operator for `_owner`, false otherwise
+	function isApprovedForAll(address _owner, address _operator) returns (bool);
+}
+
+interface ERC165 {
+    /// @notice Query if a contract implements an interface
+    /// @param interfaceID The interface identifier, as specified in ERC-165
+    /// @dev Interface identification is specified in ERC-165. This function
+    ///  uses less than 30,000 gas.
+    /// @return `true` if the contract implements `interfaceID` and
+    ///  `interfaceID` is not 0xffffffff, `false` otherwise
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+}
+```
+
+A wallet/broker/auction application MUST implement the **wallet interface** if it will accept safe transfers.
+
+```solidity
+interface ERC721TokenReceiver {
+    /// @notice Handle the receipt of a token
+    /// @dev The ERC721 smart contract calls this function on the recipient
+    ///  after a `transfer`. This function MAY throw to revert and reject the
+    ///  transfer. This function MUST use 50,000 gas or less. Return of other
+    ///  than the magic value MUST result in the transaction being reverted.
+    ///  Note: the contract address is always the message sender.
+    /// @param _from The sending address 
+    /// @param _tokenId The deed identifier which was sent
+    /// @param data Additional data with no specified format
+    /// @return Always returns `keccak256("ERC721_ONNFTRECEIVED")`, unless throwing
```

Comment:
Why not return `bytes4(keccak256("onNFTReceived(address,uint256,bytes)"))`? I don't see a point in deviating from the way selectors are already calculated.

Reply:
Thank you! Updated in e847ffe